app/js/reducers/__tests__/transaction_test.js

```diff
@@ -0,0 +1,28 @@
+jest.unmock('../transaction');
+jest.unmock('../../actions/transaction');
+
+import transaction from '../transaction';
+import { showModal, hideModal, uploadStockSuccess } from '../../actions/transaction';
+
+describe('transaction', () => {
+  it('shows modal', () => {
+    const action = showModal();
+    expect(transaction({
+      stocks: {},
+      modal: false
+    }, action)).toEqual({
```

Comment:
You do this for many, but just an idea (that may be completely off) but it may be easier to use `.to.have.property` of `expect`.

I think you can do this:
```
expect(window).to.have.property('expect')
expect(window).to.have.property('expect', expect)
expect({a: 'b'}).to.have.property('a');
```

Curious of your opinion, I can totally carry on from here and improve them as well.